trade-finance-manager-ui/server/helpers/date.test.js

```diff
@@ -0,0 +1,111 @@
+import { isSameDay } from 'date-fns';
+import { getBusinessDayOfMonth } from './date';
+
+describe('date', () => {
+  describe('getBusinessDayOfMonth', () => {
+    it.each([
+      { businessDay: 0, testCase: 'less than 1' },
+      { businessDay: 1.5, testCase: 'not an integer' },
+      { businessDay: '1', testCase: 'not a number' },
+    ])('throws when businessDay is $testCase', ({ businessDay }) => {
+      // Arrange
+      const dateInMonth = new Date('2023-01-01');
+      const holidays = [];
+
+      // Act / Assert
+      expect(() => getBusinessDayOfMonth(dateInMonth, holidays, businessDay)).toThrow(
+        new Error(`businessDay must be a positive integer. Found ${businessDay}`),
+      );
+    });
+
+    describe('when there are no holidays', () => {
+      const holidays = [];
+
+      describe('when the first of the month is not a weekend date', () => {
+        const dateInMonth = new Date('2023-11-15');
+
+        it.each([
+          { businessDay: 1, expectedResult: new Date('2023-11-01') },
+          { businessDay: 2, expectedResult: new Date('2023-11-02') },
+          { businessDay: 3, expectedResult: new Date('2023-11-03') },
+          { businessDay: 4, expectedResult: new Date('2023-11-06') },
+          { businessDay: 5, expectedResult: new Date('2023-11-07') },
+          { businessDay: 6, expectedResult: new Date('2023-11-08') },
+          { businessDay: 7, expectedResult: new Date('2023-11-09') },
+          { businessDay: 8, expectedResult: new Date('2023-11-10') },
+          { businessDay: 9, expectedResult: new Date('2023-11-13') },
+          { businessDay: 10, expectedResult: new Date('2023-11-14') },
+          { businessDay: 11, expectedResult: new Date('2023-11-15') },
+          { businessDay: 12, expectedResult: new Date('2023-11-16') },
+          { businessDay: 13, expectedResult: new Date('2023-11-17') },
+          { businessDay: 14, expectedResult: new Date('2023-11-20') },
+          { businessDay: 15, expectedResult: new Date('2023-11-21') },
+          { businessDay: 16, expectedResult: new Date('2023-11-22') },
+          { businessDay: 17, expectedResult: new Date('2023-11-23') },
+          { businessDay: 18, expectedResult: new Date('2023-11-24') },
+          { businessDay: 19, expectedResult: new Date('2023-11-27') },
+          { businessDay: 20, expectedResult: new Date('2023-11-28') },
+          { businessDay: 21, expectedResult: new Date('2023-11-29') },
+          { businessDay: 22, expectedResult: new Date('2023-11-30') },
```

Comment:
Do we really need all these test cases? Adding all these cases implies to me that we expect different behaviour when the `businessDay` goes into the next month (like throwing an error because the 23rd business day is December 1st). We don't do that as far as I can tell so I don't think we should have this many test cases